### PR TITLE
fix: width query param overrides in srcSet

### PR DESCRIPTION
**Pull request by baldurh**
## Description

This PR fixes a bug introduced in #395

The refactoring done there failed to properly override the `w` query parameter in `srcSet` if it was being passed with the src url.

### Bug Fix

- [x] All existing unit tests are still passing (if applicable)
- [x] Add new passing unit tests to cover the code introduced by your PR
- [ ] Update the readme
- [ ] Update or add any necessary API documentation
- [x] Add some [steps](#steps-to-test) so we can test your bug fix
- [x] The PR title is in the [conventional commit](#conventional-commit-spec) format: e.g. `fix(<area>): fixed bug #issue-number`
- [x] Add your info to the [contributors](#packagejson-contributors) array in package.json!

## Steps to Test

Code:

```js
<Imgix src="https://assets.imgix.net/examples/pione.jpg?w=333" />
```

v8.6.2:
```html
<img
  src="https://assets.imgix.net/examples/pione.jpg?w=333&amp;auto=format&amp;fit=crop&amp;ixlib=react-8.6.2"
  srcset="https://assets.imgix.net/examples/pione.jpg?w=333&amp;auto=format&amp;fit=crop&amp;ixlib=react-8.6.2&amp;w=100 100w, https://assets.imgix.net/examples/pione.jpg?w=333&amp;auto=format&amp;fit=crop&amp;ixlib=react-8.6.2&amp;w=116 116w, ..."
>
```

this branch:
```html
<img
  src="https://assets.imgix.net/examples/pione.jpg?w=333&amp;auto=format&amp;fit=crop&amp;ixlib=react-8.6.2"
  srcset="https://assets.imgix.net/examples/pione.jpg?auto=format&amp;fit=crop&amp;ixlib=react-8.6.2&amp;w=100 100w, https://assets.imgix.net/examples/pione.jpg?auto=format&amp;fit=crop&amp;ixlib=react-8.6.2&amp;w=116 116w, ..."
>
```
Steps:

1.  Go to https://codesandbox.io/s/elegant-joliot-i04qb
2.  Inspect the image
3.  Look at the srcSet
4. See the bug

### CODE REVIEW

File: test/unit/extract-query-params.test.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import extractQueryParams from "extractQueryParams";
+
+test("query param extraction", () => {
+  const [src, params] = extractQueryParams("https://demo.imgix.net/abc.png?w=100&auto=format,enhance")
+  expect(src).toEqual("https://demo.imgix.net/abc.png");
+  expect(params).toEqual({w: "100", auto: "format,enhance" });
+});
```
Comment (jayeb) [inline test/unit/extract-query-params.test.js]: I wouldn't mind seeing a test or two in here to check/enforce the decoding behavior that's also included in this function.

File: test/unit/react-imgix.test.jsx [hunk 2]
```diff
@@ -542,6 +542,13 @@ describe("When using the component", () => {
     expectSrcsToContain(sut, `w=${width}`);
   });
 
+  it("responsive width should overwrite the width query parameter correctly", () => {
```
Comment (jayeb) [inline test/unit/react-imgix.test.jsx]: 👍👍

File: test/unit/extract-query-params.test.js (hunk 1 above)
Comment (baldurh) [reply test/unit/extract-query-params.test.js]: Done!

**Comment by baldurh** [comment]
@jayeb I also changed the `pretty` script in `package.json` you might want to verify it ✌️

**Pull request merged**